trin-core/src/portalnet/storage.rs

```diff
@@ -649,28 +661,54 @@ pub mod test {
             sql_connection_pool,
         };
 
-        let mut storage = PortalStorage::new(storage_config)?;
+        let storage = PortalStorage::new(storage_config)?;
 
-        let value: Vec<u8> = "value".into();
+        let result = storage.find_farthest_content_id()?;
+        assert!(result.is_none());
 
-        let content_key_a = IdentityContentKey::new([1; 32]);
-        storage.store(&content_key_a, &value)?;
+        temp_dir.close()?;
+        Ok(())
+    }
+
+    #[tokio::test]
+    #[serial]
+    async fn test_find_farthest() {
+        fn prop(x: IdentityContentKey, y: IdentityContentKey) -> TestResult {
+            let temp_dir = setup_temp_dir();
+
+            let node_id = NodeId::random();
+            let val = vec![0x00, 0x01, 0x02, 0x03, 0x04];
+
+            let db = Arc::new(PortalStorage::setup_rocksdb(node_id).unwrap());
+            let sql_connection_pool = PortalStorage::setup_sql(node_id).unwrap();
+
+            let storage_config = PortalStorageConfig {
+                storage_capacity_kb: 100,
+                node_id,
+                distance_function: DistanceFunction::Xor,
+                db,
+                sql_connection_pool,
+            };
 
-        let content_key_b = IdentityContentKey::new([5; 32]);
-        storage.store(&content_key_b, &value)?;
+            let mut storage = PortalStorage::new(storage_config).unwrap();
+            storage.store(&x, &val).unwrap();
+            storage.store(&y, &val).unwrap();
 
-        let expected_content_id = content_key_b.content_id();
-        let result = storage.find_farthest_content_id()?;
-        assert_eq!(result.unwrap(), expected_content_id);
+            let expected_farthest = if storage.distance_to_content_id(&x.content_id())
+                > storage.distance_to_content_id(&y.content_id())
+            {
+                x.content_id()
+            } else {
+                y.content_id()
+            };
 
-        let content_key_c = IdentityContentKey::new([10; 32]);
-        storage.store(&content_key_c, &value)?;
+            let farthest = storage.find_farthest_content_id();
 
-        let expected_content_id = content_key_c.content_id();
-        let result = storage.find_farthest_content_id()?;
-        assert_eq!(result.unwrap(), expected_content_id);
+            temp_dir.close().unwrap();
 
-        temp_dir.close()?;
-        Ok(())
+            TestResult::from_bool(farthest.unwrap().unwrap() == expected_farthest)
+        }
+
+        quickcheck(prop as fn(IdentityContentKey, IdentityContentKey) -> TestResult);
```

Comment:
Can we use the [quickcheck! ](https://docs.rs/quickcheck/latest/quickcheck/macro.quickcheck.html)macro and simplify this?

Reply:
With the macro I'm having some issues with the derivable traits above this test function, so I think we can leave it for now. I'm also not sure that it makes the test meaningfully simpler.